### PR TITLE
feat(ui/bff): implement ConfigMap-backed CRUD for MCP catalog source configs

### PR DESCRIPTION
**Pull request by manaswinidas**
## Description

Replace the stub `McpCatalogSettingsRepository` with a real implementation that persists MCP catalog source configurations to Kubernetes ConfigMaps (`default-mcp-catalog-sources` / `mcp-catalog-sources`), mirroring the existing model catalog settings pattern.

### Changes

- Add MCP-specific ConfigMap constants (`McpCatalogSourceDefaultConfigMapName`, `McpCatalogSourceUserConfigMapName`) and K8s client interface methods (`GetAllMcpCatalogSourceConfigs`, `UpdateMcpCatalogSourceConfig`)
- Implement the new methods in `SharedClientLogic` (fetches default + user ConfigMaps, tolerates missing user CM)
- Create MCP YAML parsing helpers (`ParseMcpCatalogYaml`, `FindMcpCatalogSourceById`, `ConvertMcpSourceConfigToYamlEntry`, `UpdateMcpCatalogSourceInYAML`, etc.)
- Full repository CRUD with:
  - Merge logic for default + user sources
  - Payload validation (id format, required fields, yaml content)
  - Conflict detection via resourceVersion
  - Default source protection (cannot delete; restricted field updates)
- Update handler error mapping for new validation/forbidden/conflict errors
- Add envtest ConfigMap fixtures for MCP catalog sources
- Comprehensive Ginkgo test suite covering all CRUD operations and edge cases

## How Has This Been Tested?

- All 90 repository-level unit tests pass (`go test ./internal/repositories/ -count=1`)
- `go build ./...` and `go vet ./...` pass cleanly
- Tests use envtest with real K8s API server to verify ConfigMap CRUD, merge logic, conflict detection, and validation

## Test Impact

New test file `mcp_catalog_settings_test.go` covers:
- GetAll (merged default + user sources, isDefault flag)
- Get by ID (yaml content resolution, not-found error, isDefault)
- Create (validation: missing id/name/type/yaml, unsupported type, duplicate id, special characters; success with labels and server filters)
- Update (not-found, type change rejected, default source protections for name/labels/yaml, user source update, default override, concurrent conflict detection)
- Delete (default source protection, not-found, successful delete with yaml cleanup)

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages
- [x] Automated tests are provided as part of the PR for major new functionalities; testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).
- [ ] **For first time contributors**: Please reach out to the [Reviewers](https://github.com/kubeflow/hub/blob/main/OWNERS) to ensure all tests are being run, ensuring the label `ok-to-test` has been added to the PR.

If you have UI changes

- [x] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/bff/internal/api/mcp_catalog_settings_handler.go [hunk 1]
```diff
@@ -202,7 +206,9 @@ func (app *App) DeleteMcpCatalogSourceConfigHandler(w http.ResponseWriter, r *ht
 
 	_, err = app.repositories.McpCatalogSettingsRepository.DeleteMcpCatalogSourceConfig(ctx, client, namespace, sourceID)
```
Comment (ppadti) [inline clients/ui/bff/internal/api/mcp_catalog_settings_handler.go]: Instead of not sending the response body with just 204 status code - can we try to match the model catalog settings pattern here?

File: clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+
+	BeforeEach(func() {
+		repo = NewMcpCatalogSettingsRepository()
+		var err error
+		k8sClient, err = kubernetesMockedStaticClientFactory.GetClient(mocks.NewMockSessionContextNoParent())
+		Expect(err).NotTo(HaveOccurred())
+		ctx = mocks.NewMockSessionContextNoParent()
+	})
+
+	Describe("GetAllMcpCatalogSourceConfigs", func() {
```
Comment (ppadti) [inline clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go:29]: Missing test case for merge verification — the model catalog has "should merge, if user overrides the default source" which validates that when a user override exists for a default source, the merged result is returned in GetAll.

File: clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+		})
+
+		It("should return isDefault=false for user managed source", func() {
+			catalog, err := repo.GetMcpCatalogSourceConfig(ctx, k8sClient, "kubeflow", "custom_mcp_servers")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(*catalog.IsDefault).To(BeFalse())
+		})
+	})
+
+	Describe("CreateMcpCatalogSourceConfig", func() {
```
Comment (ppadti) [inline clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go:110]: Missing path traversal and forward slash ID validation tests. The model catalog covers these security edge cases:

"should reject catalog ID with path traversal attempt" (../../../etc/passwd)
"should reject catalog ID with forward slash" (test/malicious)

File: clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+				Type: "yaml",
+				Yaml: mcpStringPtr("servers: []"),
+			}
+			_, err := repo.CreateMcpCatalogSourceConfig(ctx, k8sClient, "kubeflow", payload)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("invalid catalog ID"))
+		})
+	})
+
+	Describe("UpdateMcpCatalogSourceConfig", func() {
```
Comment (ppadti) [inline clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go:266]: Missing 3 test cases that model catalog covers:

"should allow updating excludedServers on default catalog" — only includedServers is tested, need the symmetric excludedServers case.
"should clear includedServers when empty array is provided" — verifies that sending [] removes the field rather than being treated as a no-op.
"should clear excludedServers when empty array is provided" — same as above for excluded.

File: clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+					defaultCatalog = &c
+					break
+				}
+			}
+			Expect(defaultCatalog).NotTo(BeNil())
+			Expect(*defaultCatalog.IsDefault).To(BeTrue())
+		})
+	})
+
+	Describe("GetMcpCatalogSourceConfig", func() {
```
Comment (ppadti) [inline clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go:72]: Missing "should return merged data for user override default source config" — model catalog tests that a Get on a default source with a user override returns the merged data with correct name and isDefault=true

**Review by ppadti — CHANGES_REQUESTED** [review]
Need to add the test cases for all the cases for handler as well a for missing repository tests - could you please add that?

File: clients/ui/bff/internal/integrations/kubernetes/client.go [hunk 6]
```diff
@@ -15,6 +15,10 @@ const CatalogSourceKey = "sources.yaml"
 const CatalogSourceDefaultConfigMapName = "default-catalog-sources"
 const CatalogSourceUserConfigMapName = "model-catalog-sources"
 
+const McpCatalogSourceKey = "sources.yaml"
+const McpCatalogSourceDefaultConfigMapName = "default-mcp-catalog-sources"
```
Comment (ppadti) [inline clients/ui/bff/internal/integrations/kubernetes/client.go]: The default configMap name is wrong here, but should be "default-catalog-sources" — the same ConfigMap as model catalog defaults

File: clients/ui/bff/internal/repositories/mcp_catalog_helpers.go [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+		Catalogs []struct {
+			Name            string                 `yaml:"name"`
+			Id              string                 `yaml:"id"`
+			Type            string                 `yaml:"type"`
+			Enabled         *bool                  `yaml:"enabled"`
+			Properties      map[string]interface{} `yaml:"properties"`
+			Labels          []string               `yaml:"labels"`
+			IncludedServers []string               `yaml:"includedServers"`
+			ExcludedServers []string               `yaml:"excludedServers"`
+		} `yaml:"catalogs"`
```
Comment (ppadti) [inline clients/ui/bff/internal/repositories/mcp_catalog_helpers.go]: yaml format has to be `yaml:"mcp_catalogs"` and needs to be fixed in other 4 places in the same file

File: clients/ui/bff/internal/integrations/kubernetes/k8mocks/base_testenv.go [hunk 8]
```diff
@@ -1417,6 +1437,99 @@ func createTransferJobPodEvents(k8sClient kubernetes.Interface, ctx context.Cont
...
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			k8s.McpCatalogSourceKey:      raw,
+			"community_mcp_servers.yaml": "servers:\n - name: community_server_1",
+		},
+	}
+
+	if _, err := k8sClient.CoreV1().ConfigMaps(namespace).Create(ctx, cm, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("failed to create default-mcp-catalog-sources configmap: %w", err)
```
Comment (ppadti) [inline clients/ui/bff/internal/integrations/kubernetes/k8mocks/base_testenv.go]: This needs to be changed accordingly to `default-catalog-sources`

File: clients/ui/bff/internal/integrations/kubernetes/k8mocks/base_testenv.go [hunk 9]
```diff
@@ -1417,6 +1437,99 @@ func createTransferJobPodEvents(k8sClient kubernetes.Interface, ctx context.Cont
 	return nil
 }
 
+func createMcpCatalogDefaultSourcesConfigMap(
```
Comment (ppadti) [inline clients/ui/bff/internal/integrations/kubernetes/k8mocks/base_testenv.go]: We don't need to create new default configMap because - we are using the same default-catalog-sources configMap. So we should merge this data into existing configMap

File: clients/ui/bff/internal/integrations/kubernetes/k8mocks/base_testenv.go [hunk 10]
```diff
@@ -1417,6 +1437,99 @@ func createTransferJobPodEvents(k8sClient kubernetes.Interface, ctx context.Cont
...
+	return nil
+}
+
+func createMcpCatalogSourcesConfigMap(
+	k8sClient kubernetes.Interface,
+	ctx context.Context,
+	namespace string,
+) error {
+	raw := strings.TrimSpace(`
+catalogs:
```
Comment (ppadti) [inline clients/ui/bff/internal/integrations/kubernetes/k8mocks/base_testenv.go]: This createMcpCatalogSourcesConfigMap uses `catalogs:` here but should use `mcp_catalogs:`

File: clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go [hunk 11]
```diff
@@ -0,0 +1,488 @@
...
+				Name: "Updated by Request A",
+			}
+			_, err = repo.UpdateMcpCatalogSourceConfig(ctx, k8sClient, "kubeflow", "mcp_concurrency_test", payloadA)
+			Expect(err).NotTo(HaveOccurred())
+
+			_, cmAfterA, err := k8sClient.GetAllMcpCatalogSourceConfigs(ctx, "kubeflow")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cmAfterA.ResourceVersion).NotTo(Equal(resourceVersionA))
+
+			cmB.Data[k8s.McpCatalogSourceKey] = "catalogs:\n  - id: mcp_concurrency_test\n    name: Updated by Request B\n    type: yaml\n    enabled: true"
```
Comment (ppadti) [inline clients/ui/bff/internal/repositories/mcp_catalog_settings_test.go]: These test also need to be updated to have `mcp_catalogs:` instead of `catalogs"`

**Review by ppadti — CHANGES_REQUESTED** [review]

**Review by ppadti — CHANGES_REQUESTED** [review]
The preview endpoint is not working as expected
we need to update `CatalogSourcePreviewRequest` struct to be 
```
type CatalogSourcePreviewRequest struct {
	Type            string                 `json:"type"`
	IncludedModels  []string               `json:"includedModels,omitempty"`
	ExcludedModels  []string               `json:"excludedModels,omitempty"`
	IncludedServers []string               `json:"includedServers,omitempty"`
	ExcludedServers []string               `json:"excludedServers,omitempty"`
	Properties      map[string]interface{} `json:"properties,omitempty"`
}
```
and need to update the `CreateCatalogSourcePreview` according to send IncludedServers and ExcludedServers

**Review by ppadti — APPROVED** [review]
/lgtm
/approve
